arangod/Agency/Node.cpp

```diff
@@ -141,83 +104,68 @@ std::string Node::uri() const {
 }
 
 /// @brief Move constructor
-Node::Node(Node&& other)
-    : _nodeName(std::move(other._nodeName)),
-      _parent(nullptr),
-      _store(nullptr),
-      _children(std::move(other._children)),
-      _ttl(std::move(other._ttl)),
-      _value(std::move(other._value)),
-      _vecBuf(std::move(other._vecBuf)),
-      _vecBufDirty(std::move(other._vecBufDirty)),
-      _isArray(std::move(other._isArray)) {
-  // The _children map has been moved here, therefore we must
-  // correct the _parent entry of all direct children:
-  for (auto& child : _children) {
-    child.second->_parent = this;
-  }
+Node::Node(Node&& other) : _parent(nullptr), _store(nullptr) {
+  *this = std::move(other);
```

Comment:
The operator= is public but does not touch the _parent and _store, is this a good idea?